workflow/controller/operator.go

```diff
@@ -1037,6 +1037,36 @@ func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, node *wfv1.NodeStatu
 			node.Phase = wfv1.NodeError
 		} else {
 			node.Outputs = &outputs
+			tmplCtx, err := woc.createTemplateContext(node.GetTemplateScope())
+			if err != nil {
+				log.Errorf("Failed to create template context for node %s", node.ID)
+				return nil
+			}
+			_, resolvedTmpl, _, err := tmplCtx.ResolveTemplate(node)
+			if err != nil {
+				log.Errorf("Failed to resolve template for node %s: %s", node.ID, err)
+				return nil
+			}
+
+			localParams := make(map[string]string)
```

Comment:
what's all this about @simster7 ?

Reply:
This does Argo variable substitution. @rbreeze not sure why this is here?

Reply:
The `localParams` here are for resolving the template a few lines later, which is necessary to check if the template has memoization enabled or not

Reply:
@simster7 I just remembered that this is relevant for processing template args, i.e. injecting cache key variables. So I think this section will still be necessary even after implementing the `Memoized` var in `NodeType` . I'll investigate a better location for it